web/src/components/LoginModal/LoginModal.js

```diff
@@ -0,0 +1,189 @@
+import { useState } from 'react'
+import Dialog from '@material-ui/core/Dialog'
+import Tab from '@material-ui/core/Tab'
+import Tabs from '@material-ui/core/Tabs'
+import InputTextForm from 'src/components/InputTextForm'
+import OutBound from 'src/components/OutBound'
+import { Form, Submit } from '@redwoodjs/forms'
+import { useAuth } from '@redwoodjs/auth'
+import { useFlash } from '@redwoodjs/web'
+import { Link, routes } from '@redwoodjs/router'
+import { subscribe } from 'src/helpers/subscribe'
+
+const LoginModal = ({ open, onClose, shouldStartWithSignup = false }) => {
+  const { logIn, signUp } = useAuth()
+  const { addMessage } = useFlash()
+
+  const [tab, setTab] = useState(shouldStartWithSignup ? 0 : 1)
+  const onTabChange = (_, newValue) => {
+    setTab(newValue)
+    setError('')
+  }
+  const [checkBox, setCheckBox] = useState(true)
+  const [error, setError] = useState('')
+
+  const onSubmitSignUp = async ({ email, password, name, userName }) => {
+    try {
+      setError('')
+      if (checkBox) {
+        subscribe({ email, addMessage })
```

Comment:
Somewhat crude way of subscribing the user to the newsletter, this doesn't got through the API like everything else so it's a little out place. But it works for now.